libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -276,366 +288,358 @@ const AP_Param::GroupInfo Compass::var_info[] = {
     // @User: Advanced
     // @Calibration: 1
 
-    // @Param: DIA3_Y
+    // @Param: _DIA3_Y
     // @DisplayName: Compass3 soft-iron diagonal Y component
     // @Description: DIA_Y in the compass3 soft-iron calibration matrix: [[DIA_X, ODI_X, ODI_Y], [ODI_X, DIA_Y, ODI_Z], [ODI_Y, ODI_Z, DIA_Z]]
     // @User: Advanced
     // @Calibration: 1
 
-    // @Param: DIA3_Z
+    // @Param: _DIA3_Z
     // @DisplayName: Compass3 soft-iron diagonal Z component
     // @Description: DIA_Z in the compass3 soft-iron calibration matrix: [[DIA_X, ODI_X, ODI_Y], [ODI_X, DIA_Y, ODI_Z], [ODI_Y, ODI_Z, DIA_Z]]
     // @User: Advanced
-    AP_GROUPINFO("DIA3",    28, Compass, _state._priv_instance[2].diagonals, 0),
 
-    // @Param: ODI3_X
+    // index 28
+
+    // @Param: _ODI3_X
     // @DisplayName: Compass3 soft-iron off-diagonal X component
     // @Description: ODI_X in the compass3 soft-iron calibration matrix: [[DIA_X, ODI_X, ODI_Y], [ODI_X, DIA_Y, ODI_Z], [ODI_Y, ODI_Z, DIA_Z]]
     // @User: Advanced
     // @Calibration: 1
 
-    // @Param: ODI3_Y
+    // @Param: _ODI3_Y
     // @DisplayName: Compass3 soft-iron off-diagonal Y component
     // @Description: ODI_Y in the compass3 soft-iron calibration matrix: [[DIA_X, ODI_X, ODI_Y], [ODI_X, DIA_Y, ODI_Z], [ODI_Y, ODI_Z, DIA_Z]]
     // @User: Advanced
     // @Calibration: 1
 
-    // @Param: ODI3_Z
+    // @Param: _ODI3_Z
     // @DisplayName: Compass3 soft-iron off-diagonal Z component
     // @Description: ODI_Z in the compass3 soft-iron calibration matrix: [[DIA_X, ODI_X, ODI_Y], [ODI_X, DIA_Y, ODI_Z], [ODI_Y, ODI_Z, DIA_Z]]
     // @User: Advanced
-    AP_GROUPINFO("ODI3",    29, Compass, _state._priv_instance[2].offdiagonals, 0),
-#endif // COMPASS_MAX_INSTANCES
+
+    // index 29
 
 #if COMPASS_CAL_ENABLED
-    // @Param: CAL_FIT
+    // @Param: _CAL_FIT
     // @DisplayName: Compass calibration fitness
     // @Description: This controls the fitness level required for a successful compass calibration. A lower value makes for a stricter fit (less likely to pass). This is the value used for the primary magnetometer. Other magnetometers get double the value.
     // @Range: 4 32
     // @Values: 4:Very Strict,8:Strict,16:Default,32:Relaxed
     // @Increment: 0.1
     // @User: Advanced
-    AP_GROUPINFO("CAL_FIT", 30, Compass, _calibration_threshold, AP_COMPASS_CALIBRATION_FITNESS_DEFAULT),
+    AP_GROUPINFO("_CAL_FIT", 30, Compass, _calibration_threshold, AP_COMPASS_CALIBRATION_FITNESS_DEFAULT),
 #endif
 
 #ifndef HAL_BUILD_AP_PERIPH
-    // @Param: OFFS_MAX
+    // @Param: _OFFS_MAX
     // @DisplayName: Compass maximum offset
     // @Description: This sets the maximum allowed compass offset in calibration and arming checks
     // @Range: 500 3000
     // @Increment: 1
     // @User: Advanced
-    AP_GROUPINFO("OFFS_MAX", 31, Compass, _offset_max, AP_COMPASS_OFFSETS_MAX_DEFAULT),
+    AP_GROUPINFO("_OFFS_MAX", 31, Compass, _offset_max, AP_COMPASS_OFFSETS_MAX_DEFAULT),
```

Comment:
unrelated to this PR but this parameter should probably be named "_OFS_MAX" (no double F)